[dtensor] redistributed by default take self mesh instead

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #99062
* #98452
* #99061
* __->__ #99060

This PR switches redistribute to default use self mesh instead of
the global mesh, which is more user friendly

wanchaol committed:
[dtensor] redistributed by default take self mesh instead

This PR switches redistribute to default use self mesh instead of
the global mesh, which is more user friendly

[ghstack-poisoned]

wanchaol committed:
Update on "[dtensor] redistributed by default take self mesh instead"


This PR switches redistribute to default use self mesh instead of
the global mesh, which is more user friendly

[ghstack-poisoned]

kumpera commented on torch/distributed/_tensor/api.py:
```diff
@@ -338,7 +338,7 @@ def redistribute(
         # of place redistribution, i.e. it always create a new
         # DTensor object and leave the original one unchanged.
-        device_mesh = get_global_device_mesh() if device_mesh is None else device_mesh
+        device_mesh = self.device_mesh if device_mesh is None else device_mesh
```
the more idiomatic way is `device_mesh or self.device_mesh`

wanchaol committed:
Update on "[dtensor] redistributed by default take self mesh instead"


This PR switches redistribute to default use self mesh instead of
the global mesh, which is more user friendly

[ghstack-poisoned]
